llvm.sh

```diff
@@ -94,13 +88,54 @@ case ${DISTRO} in
         ;;
 esac
 
+while getopts ":hm:n:" arg; do
```

Comment:
This is having fragmented argument parsing all over the place, putting this before setting `BASE_URL` if you could, its better for maintenance...

Reply:
The flow I thought is to initialize those variables with code above and override them if any argument passed, so with that it might be hard to move it above initialization. However it is always a good idea to add some more comments if this was kept.

Reply:
There is parsing of "version" and "all" above though...